lib/versioner/files_counter.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2017 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package versioner
+
+import (
+	"path/filepath"
+	"sort"
+
+	"github.com/syncthing/syncthing/lib/fs"
+)
+
+// filesCounter is intended to track empty directories
+type filesCounter struct {
+	dirs map[string]struct{}
+}
```

Comment:
The previous comments still apply:
The type can simply be
```
type filesCounter map[string]struct{}
```

And it should probably be renamed, as it is no longer counting files (e.g. "emptyDirTracker").

Reply:
Done. Naming is hard so I picked up your variant.